src/package.ts

```diff
@@ -1840,6 +1843,20 @@ export async function pack(options: IPackageOptions = {}): Promise<IPackageResul
 	return { manifest, packagePath, files };
 }
 
+export async function signPackage(packageFile: string, signScript: string): Promise<string> {
+	const packageFolder = path.dirname(packageFile);
+	const packageName = path.basename(packageFile, '.vsix');
+	const manifestFile = path.join(packageFolder, `${packageName}.signature.manifest`);
+	const signatureFile = path.join(packageFolder, `${packageName}.signature.p7s`);
+	const signatureZip = path.join(packageFolder, `${packageName}.signature.zip`);
+
+	await generateManifest(packageFile, manifestFile);
+	const { stdout } = await promisify(cp.execFile)(signScript, [manifestFile,  signatureFile]);
+	console.log(stdout);
```

Comment:
Another idea, since using sync is more than fine here:

```suggestion
	await cp.spawnSync(signScript, [manifestFile,  signatureFile], { stdio: 'inherit' });
```